src/ASM/LR/ASMu3Drecovery.C

```diff
@@ -533,6 +535,9 @@ bool ASMu3D::faceL2projection (const DirichletFace& face,
   for (size_t ie = 0; ie < face.MLGE.size(); ie++) // for all face elements
   {
     int iel = 1 + face.MLGE[ie];
+    int ielG = iel;
+    if (geo != lrspline.get())
+      ielG = geo->getElementContaining(lrspline->getElement(iel-1)->midpoint()) + 1;
```

Comment:
Here also, consider the shorter `int ielG = geo == lrspline.get() ? iel : ...;` syntac although the line then gets even longer. But it is already beyond the 80-char limit, afaict

Reply:
it get's obnoxiously long but okay.